### PR TITLE
uart_set_baudrate should return actual rate set even in case of out o…

### PR DESCRIPTION
**Pull request by kilograham**
…f range parameters

### CODE REVIEW

File: src/common/pico_base/include/pico/error.h [hunk 1]
```diff
@@ -16,6 +16,7 @@ enum {
     PICO_ERROR_TIMEOUT = -1,
     PICO_ERROR_GENERIC = -2,
     PICO_ERROR_NO_DATA = -3,
+    PICO_ERROR_OUT_OF_RANGE = -4,
```
Comment (lurch) [inline src/common/pico_base/include/pico/error.h:19]: Looks like you've added this constant but not actually used it anywhere? :confused:

**Review by Wren6991 — COMMENTED** [review]
Nice, I took it for a spin
<details>
<summary>Log</summary>
Attempting 2000, got 2000
Attempting 1950, got 1950
Attempting 1900, got 1899
Attempting 1850, got 1850
Attempting 1800, got 1799
Attempting 1750, got 1750
Attempting 1700, got 1699
Attempting 1650, got 1650
Attempting 1600, got 1600
Attempting 1550, got 1549
Attempting 1500, got 1500
Attempting 1450, got 1449
Attempting 1400, got 1399
Attempting 1350, got 1350
Attempting 1300, got 1300
Attempting 1250, got 1250
Attempting 1200, got 1199
Attempting 1150, got 1149
Attempting 1100, got 1100
Attempting 1050, got 1050
Attempting 1000, got 1000
Attempting 950, got 949
Attempting 900, got 899
Attempting 850, got 850
Attempting 800, got 800
Attempting 750, got 749
Attempting 700, got 699
Attempting 650, got 649
Attempting 600, got 600
Attempting 550, got 549
Attempting 500, got 500
Attempting 450, got 450
Attempting 400, got 400
Attempting 350, got 350
Attempting 300, got 299
Attempting 250, got 250
Attempting 200, got 200
Attempting 150, got 150
Attempting 100, got 119
Attempting 50, got 119
</details>

Other than the stray error number, happy to merge this

**Comment by lurch** [comment]
I checked the [header](https://github.com/raspberrypi/pico-sdk/blob/master/src/rp2_common/hardware_uart/include/hardware/uart.h#L155) to see if the doxygen-comment is still correct, and it is :slightly_smiling_face:
"Set baud rate as close as possible to requested, and return actual rate selected."

However would you mind adding "\return Actual set baudrate", so that it matches `uart_init` please?

File: src/rp2_common/hardware_uart/uart.c [hunk 2]
```diff
@@ -75,23 +75,25 @@ uint uart_set_baudrate(uart_inst_t *uart, uint baudrate) {
     uint32_t baud_rate_div = (8 * clock_get_hz(clk_peri) / baudrate);
     uint32_t baud_ibrd = baud_rate_div >> 7;
     uint32_t baud_fbrd = ((baud_rate_div & 0x7f) + 1) / 2;
-    invalid_params_if(UART, (baud_ibrd > 65535) || (baud_ibrd == 0));
```
Comment (lurch) [inline src/rp2_common/hardware_uart/uart.c:78]: I'm not sure how these macros are intended to be used, but is it "consistent" that this `invalid_params_if` has been removed, but the `invalid_params_if(UART, baudrate == 0);` is still there on line 74? :shrug:

Comment (kilograham) [reply src/rp2_common/hardware_uart/uart.c:78]: ha i didn't notice that

Comment (Wren6991) [reply src/rp2_common/hardware_uart/uart.c:78]: 0 is invalid because we would rather not divide by it :)

**Review by Wren6991 — APPROVED** [review]

**Pull request merged**